protected/humhub/modules/file/models/File.php

```diff
@@ -126,9 +179,30 @@ public function getHistoryFiles()
      */
     public function beforeSave($insert)
     {
+        /**
+         * Used for file history
+         *
+         * @see FileHistory::createEntryForFile()
+         */
         $this->old_updated_by = $this->getOldAttribute('updated_by');
         $this->old_updated_at = $this->getOldAttribute('updated_at');
 
+        if ($this->category === null) {
```

Comment:
I would leave that out for the beta for now.
But I like the idea, to use the category to store the file type.